### PR TITLE
test(bridge): cover escape_forwarded_host and forwarded_for_value

### LINKED ISSUE #274: test(bridge): add edge-case tests for escape_forwarded_host and forwarded_for_value
**Crate:** `spooky-bridge` | **File:** `crates/bridge/src/forwarded.rs` (`forwarded_for_value:102`, `escape_forwarded_host:109`)

### Description

`escape_forwarded_host` (`crates/bridge/src/forwarded.rs:109`) and `forwarded_for_value` (`forwarded.rs:102`) are helpers used in every proxied request's `Forwarded` header. Neither has dedicated tests, leaving RFC 7239 compliance unverified. The only existing coverage is indirect: IPv6 formatting via `h3_to_h2.rs:424` and host quoting via `h3_to_h2.rs:274`; no test feeds a host containing `"` or `\` to exercise the escaping path.

### Steps / Context

- `escape_forwarded_host`: test hosts containing `"` and `\` (must be escaped as `\"` and `\\`)
- `forwarded_for_value`: IPv4 → bare string, IPv6 → `"[addr]"` quoted form
- Tests are best placed in a new `#[cfg(test)]` block in `forwarded.rs` (these are pure functions)

### Expected vs Actual

**Expected:** Special characters in host are escaped; IPv6 is quoted per RFC 7239 §6.
**Actual:** No dedicated tests — an escaping bug would produce malformed `Forwarded` headers that downstream proxies may misparse.

### Impact

Tests only. Pure string logic — no async, no network. Easy to verify against the RFC.

### PR DESCRIPTION
**Pull request by kevinnft**
## Summary
Closes #274 and #267.

Adds unit tests for Forwarded header helpers used on every proxied request:

### #274 — pure helpers
- `escape_forwarded_host` — escapes `"` and `\`
- `forwarded_for_value` — IPv4 bare / IPv6 quoted `[addr]`

### #267 — policy modes
- `merge_forwarded_chain` for Preserve / Append / Overwrite
- `build_forwarded_header_values` Overwrite drops inbound
- Append keeps inbound XFF chain
- Preserve keeps inbound only

## Test plan
- [x] Local pure-function checks for escape + IP formatting
- [ ] `cargo test -p spooky-bridge` in CI

### CODE REVIEW

**Review by rudraditya21 — APPROVED** [review]

**Comment by rudraditya21** [comment]
Hi! Thanks for the contribution. It looks like the CI is failing because of a formatting issue. Could you please run:
```
cargo fmt --all
```
Then commit the formatting changes and push the updated branch. That should resolve the cargo fmt --all -- --check failure. Thanks!

**Comment by kevinnft** [comment]
Ran `cargo fmt --all` and pushed — should clear the lint/fmt CI failure. Thanks for the pointer!

File: crates/bridge/src/forwarded.rs [hunk 1]
```diff
@@ -109,3 +109,128 @@ pub fn forwarded_for_value(ip: IpAddr) -> String {
...
+        };
+        let ip = IpAddr::V4(Ipv4Addr::new(203, 0, 113, 10));
+        let out =
+            build_forwarded_header_values(&policy, inbound, ip, "app.example").expect("build");
+
+        assert_eq!(out.x_forwarded_for.unwrap().as_bytes(), b"203.0.113.10");
+        assert_eq!(out.x_forwarded_proto.unwrap().as_bytes(), b"https");
+        assert_eq!(out.x_forwarded_host.unwrap().as_bytes(), b"app.example");
+
+        let s = std::str::from_utf8(out.forwarded.unwrap().as_bytes()).unwrap();
```
Comment (nishujangra) [inline crates/bridge/src/forwarded.rs]: Failure with error:

```sh
error[E0716]: temporary value dropped while borrowed
Error:    --> crates/bridge/src/forwarded.rs:195:37
    |
195 |         let s = std::str::from_utf8(out.forwarded.unwrap().as_bytes()).unwrap();
    |                                     ^^^^^^^^^^^^^^^^^^^^^^                     - temporary value is freed at the end of this statement
    |                                     |
    |                                     creates a temporary value which is freed while still in use
196 |         assert!(s.contains("for=203.0.113.10"));
    |                 - borrow later used here
    |
help: consider using a `let` binding to create a longer lived value
    |
195 ~         let binding = out.forwarded.unwrap();
196 ~         let s = std::str::from_utf8(binding.as_bytes()).unwrap();
```

**Comment by kevinnft** [comment]
Fixed the E0716 from the review: bind `out.forwarded.unwrap()` into a local before `from_utf8(...as_bytes())` so the temporary isn't dropped while `s` is borrowed. Pushed in 44ffbf1.

Local full `cargo test` for the bridge crate is blocked here by boring-sys/cmake sandbox perms, but the change matches the rustc suggestion from the lint log.
